hikari/impl/event_manager_base.py

```diff
@@ -272,7 +272,11 @@ async def wait_for(
         pair = (predicate, future)
 
         waiter_set.add(pair)  # type: ignore[arg-type]
-        return await asyncio.wait_for(future, timeout=timeout)
+        try:
+            return await asyncio.wait_for(future, timeout=timeout)
+        except Exception:
+            waiter_set.remove(pair)  # type: ignore[arg-type]
```

Comment:
i assume this won't be removed anywhere else right?

Reply:
It does, so changed it so that its only removed on Timeout instead of all exceptions.